shell/platform/darwin/ios/framework/Source/FlutterKeyResponderTest.mm

```diff
@@ -0,0 +1,1137 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Foundation/Foundation.h>
+#import <OCMock/OCMock.h>
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterKeyResponder.h"
+#import "flutter/testing/testing.h"
+#include "third_party/googletest/googletest/include/gtest/gtest.h"
+
+// A wrap to convert FlutterKeyEvent to a ObjC class.
+@interface TestKeyEvent : NSObject
+@property(nonatomic) FlutterKeyEvent* data;
+@property(nonatomic) FlutterKeyEventCallback callback;
+@property(nonatomic) _VoidPtr userData;
+- (nonnull instancetype)initWithEvent:(const FlutterKeyEvent*)event
+                             callback:(nullable FlutterKeyEventCallback)callback
+                             userData:(nullable _VoidPtr)userData;
+- (BOOL)hasCallback;
+- (void)respond:(BOOL)handled;
+@end
+
+@implementation TestKeyEvent
+- (instancetype)initWithEvent:(const FlutterKeyEvent*)event
+                     callback:(nullable FlutterKeyEventCallback)callback
+                     userData:(nullable _VoidPtr)userData {
+  self = [super init];
+  _data = new FlutterKeyEvent(*event);
+  if (event->character != nullptr) {
+    size_t len = strlen(event->character);
+    char* character = new char[len + 1];
+    strcpy(character, event->character);
+    _data->character = character;
+  }
+  _callback = callback;
+  _userData = userData;
+  return self;
+}
+
+- (BOOL)hasCallback {
+  return _callback != nil;
+}
+
+- (void)respond:(BOOL)handled {
+  NSAssert(
+      _callback != nil,
+      @"Improper call to `respond` that does not have a callback.");  // Caller's responsibility
+  _callback(handled, _userData);
+}
+
+- (void)dealloc {
+  if (_data->character != nullptr)
+    delete[] _data->character;
+  delete _data;
+}
+@end
+
+namespace flutter::testing {
+
+namespace {
+constexpr uint64_t kKeyCodeKeyA = 0x00;
+constexpr uint64_t kKeyCodeKeyW = 0x0d;
+constexpr uint64_t kKeyCodeShiftLeft = 0x38;
+constexpr uint64_t kKeyCodeShiftRight = 0x3c;
+constexpr uint64_t kKeyCodeCapsLock = 0x39;
+constexpr uint64_t kKeyCodeNumpad1 = 0x53;
+constexpr uint64_t kKeyCodeF1 = 0x7a;
+constexpr uint64_t kKeyCodeAltRight = 0x3d;
+
+constexpr uint64_t kPhysicalKeyA = 0x00070004;
+constexpr uint64_t kPhysicalKeyW = 0x0007001a;
+constexpr uint64_t kPhysicalShiftLeft = 0x000700e1;
+constexpr uint64_t kPhysicalShiftRight = 0x000700e5;
+constexpr uint64_t kPhysicalCapsLock = 0x00070039;
+constexpr uint64_t kPhysicalNumpad1 = 0x00070059;
+constexpr uint64_t kPhysicalF1 = 0x0007003a;
+constexpr uint64_t kPhysicalAltRight = 0x000700e6;
+
+constexpr uint64_t kLogicalKeyA = 0x00000061;
+constexpr uint64_t kLogicalKeyW = 0x00000077;
+constexpr uint64_t kLogicalShiftLeft = 0x0030000010d;
+constexpr uint64_t kLogicalShiftRight = 0x0040000010d;
+constexpr uint64_t kLogicalCapsLock = 0x00000000104;
+constexpr uint64_t kLogicalNumpad1 = 0x00200000031;
```

Comment:
These values have been changed with the latest codegen change. See https://github.com/flutter/engine/pull/25883.